sql/analyze_test.go

```diff
@@ -207,11 +207,10 @@ func TestSimplifyExpr(t *testing.T) {
 		{`i LIKE 'foo%'`, `i >= 'foo' AND i < 'fop'`},
 		{`i LIKE 'foo_'`, `i >= 'foo' AND i < 'fop'`},
 		{`i LIKE 'bar_foo%'`, `i >= 'bar' AND i < 'bas'`},
-		// TODO(pmattis): unsupported comparison operator: SIMILAR TO
-		// {`i SIMILAR TO '.*'`, `true`},
-		// {`i SIMILAR TO 'foo'`, `i = 'foo'`},
-		// {`i SIMILAR TO 'foo.*'`, `i >= 'foo' AND i < 'fop'`},
-		// {`i SIMILAR TO '(foo|foobar).*'`, `i >= 'foo' AND i < 'fop'`},
+		{`i SIMILAR TO '.*'`, `true`},
+		{`i SIMILAR TO 'foo'`, `i = 'foo'`},
+		{`i SIMILAR TO 'foo.*'`, `i >= 'foo' AND i < 'fop'`},
+		{`i SIMILAR TO '(foo|foobar).*'`, `i >= 'foo' AND i < 'fop'`},
```

Comment:
The code in `simplifyComparisonExpr` is going to need to be changed. I think we'll want the `parser` package to export a function to normalize the SQL-style regex syntax to that expected by the `regexp` package.